mbedtls/build.rs

```diff
@@ -6,32 +6,53 @@
  * option. This file may not be copied, modified, or distributed except
  * according to those terms. */
 
-use std::collections::{HashMap, HashSet};
-use std::env;
+ use std::collections::{HashMap, HashSet};
+ use std::env;
+ 
+ use rustc_version::Channel;
 
-use rustc_version::Channel;
+ use std::collections::hash_map::DefaultHasher;
+ use std::hash::{Hasher};
+ 
 
-/// Return the crate hash that Cargo will be passing to `rustc -C metadata=`.
-// If there's a panic in this code block, that means Cargo's way of running the
-// build script has changed, and this code should be updated to handle the new
-// case.
-fn get_compilation_metadata_hash() -> String {
+/// Retrieves or generates a metadata value used for symbol name mangling to ensure unique C symbols.
+/// When building with Cargo, the metadata value is extracted from the OUT_DIR environment variable. 
+/// For Bazel builds, this method approximates Cargo's -Cmetadata by hashing the crate name and version, 
+/// which are sufficient for ensuring symbol uniqueness.
+fn get_compilation_symbol_suffix() -> String {
     let out_dir: std::path::PathBuf = std::env::var_os("OUT_DIR").unwrap().into();
     let mut out_dir_it = out_dir.iter().rev();
-    assert_eq!(out_dir_it.next().unwrap(), "out");
-    let crate_ = out_dir_it.next().unwrap().to_string_lossy();
-    assert!(crate_.starts_with("mbedtls-"));
-    crate_[8..].to_owned()
+    let next = out_dir_it.next().unwrap();
+
+    if next == "out" {
+        // If Cargo is used as build system.
+        let crate_ = out_dir_it.next().unwrap().to_string_lossy();
+        assert!(crate_.starts_with("mbedtls-"));
+        return crate_[8..].to_owned();
+    } else if next == "_bs.out_dir" {
+        // If Bazel is the build system.
+        let package_name = std::env::var("CARGO_PKG_NAME").expect("package name missed");
+        let version = std::env::var("CARGO_PKG_VERSION").expect("Package version missed");
```

Comment:
Nit: `missed` --> `missing`
Also, package is spelled once with an uppercase p and once with a lowercase p.